lib/db/mem.js

```diff
@@ -1307,13 +1310,19 @@ module.exports = function (log, error) {
 
   Memory.prototype.replaceRecoveryCodes = function (uid, count) {
     uid = uid.toString('hex')
+    let codes
     return getAccountByUid(uid)
       .then(() => {
-        return dbUtil.generateRecoveryCodes(count)
-          .then((codes) => {
-            recoveryCodes[uid] = codes.map((code) => {
+        return dbUtil.generateRecoveryCodes(count, RECOVERY_CODE_KEYSPACE, RECOVERY_CODE_LENGTH)
+          .then((result) => {
+            codes = result
+            return Promise.all(codes.map((code) => dbUtil.createHashScrypt(code)))
+          })
+          .then((hashes) => {
```

Comment:
I might be misremembering, but does bluebird have a `.all` method we could used here instead of returning an explicit `Promise.all` from the previous `then`?